### PR TITLE
[fix] Fixed extensibility of openwisp-users

### PR DESCRIPTION
**Pull request by pandafy**
Need for this PR arose from https://github.com/openwisp/openwisp-controller/pull/390 where some bugs hurting extensibility of openwisp-users were found.

### CODE REVIEW

File: openwisp_notifications/migrations/0003_notificationsetting.py
Comment (pandafy) [inline openwisp_notifications/migrations/0003_notificationsetting.py:18]: Let's move it to `migrations/__init__.py`, since if might be required and used in other files too.

File: openwisp_notifications/handlers.py
Comment (pandafy) [inline openwisp_notifications/handlers.py:68]:
```suggestion
    user_app_name = User._meta.app_label
```

We are already loading `User` and `Organization` model using swapper. I don't think we need to do that again here.

**Review by pandafy — COMMENTED** [review]
:warning:  **Requested Changes**

@devkapilbansal let's just fix the imports in this PR and remove all hard coded reference to openwisp-users. 

Removing a migration file introduces a change migration history which can affect existing installations of openwisp-notifications. I have opened https://github.com/openwisp/openwisp-notifications/issues/172 for that, we will come back to it later.

File: openwisp_notifications/migrations/0002_default_permissions.py
Comment (devkapilbansal) [inline openwisp_notifications/migrations/0002_default_permissions.py:66]: @pandafy I added this code instead of removing the migration.

**Review by nemesisdesign — COMMENTED** [review]
Looks good, tests pass, @devkapilbansal have you tested this with the other PR which was failing? Does it solve the issue? Can you update the other PR to point to this branch please?

@pandafy what do you think?

**Comment by devkapilbansal** [comment]
> Looks good, tests pass, @devkapilbansal have you tested this with the other PR which was failing? Does it solve the issue? Can you update the other PR to point to this branch please?

Yes, this solves the issue. Tests passes [here](https://github.com/openwisp/openwisp-controller/pull/390) and we are using this branch there

**Comment by devkapilbansal** [comment]
However, I am not sure if this is the right approach to do so

**Review by nemesisdesign — COMMENTED** [review]
> > Looks good, tests pass, @devkapilbansal have you tested this with the other PR which was failing? Does it solve the issue? Can you update the other PR to point to this branch please?
> 
> Yes, this solves the issue. Tests passes [here](https://github.com/openwisp/openwisp-controller/pull/390) and we are using this branch there

That's great @devkapilbansal, thank you! I reviewed the other PRs and asked for some minor improvements.

This one looks fine and will wait for the greenlight of @pandafy to merge.

**Review by pandafy — COMMENTED** [review]
I sent some minor improvements in https://github.com/openwisp/openwisp-notifications/pull/173/commits/5fa6601f78a11129ab996c697a3f9242e5d7ce94. 

LGTM 👍🏼 
We can merge it!

**Review by nemesisdesign — APPROVED** [review]
Thanks @devkapilbansal @pandafy

**Pull request merged**
